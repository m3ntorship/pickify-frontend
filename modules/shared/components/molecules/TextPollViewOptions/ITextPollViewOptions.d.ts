declare namespace ITextPollViewOptions {
  export interface IProps {
    optionsGroups: IOptionGroups;
  }

  export interface IOption {
    id: string;
    body: string;
  }
  export interface IGroup {
    id: string;
    options: IOptions[];
    name: string;
  }

  export interface IOptionGroups {
    groups: IGroup[];
  }
}

export { ITextPollViewOptions };
