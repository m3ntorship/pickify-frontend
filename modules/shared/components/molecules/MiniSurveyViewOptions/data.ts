export const optionsGroups = {
  groups: [
    {
      id: '03644270-7171-4147-b5a1-4233ff547f7a',
      name: 'Group Name',
      media: [],
      options: [
        {
          id: 'eeb0a32a-1a3c-4c7f-9e50-dd3d3f70b53d',
          body: 'ali tramsy',
          media: [{ url: 'https://placeimg.com/640/480/any' }],
          vote_count: 10,
          voted: false,
        },
        {
          id: 'eeb0a32a-1a3c-4c7f-9e50-dd3d3f70b5aa',
          body: ' ahmed essam',
          media: [{ url: 'https://placeimg.com/640/480/any' }],
          vote_count: 20,
          voted: false,
        },
        {
          id: 'eeb0a32a-1a3c-4c7f-9e50-dd3d3f70b53c',
          body: ' omar gamal',
          media: [{ url: 'https://placeimg.com/640/480/any' }],
          vote_count: 30,
          voted: false,
        },
      ],
    },
  ],
};
