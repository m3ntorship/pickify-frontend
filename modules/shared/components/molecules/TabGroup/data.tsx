import React from 'react';
import { EPostType } from '../../../types/postFeed/EPostType';
import ImagePoll from '../../icons/imagePoll.svg';
import TextPoll from '../../icons/textPoll.svg';
import MiniSurvey from '../../icons/miniSurvey.svg';
import styles from '../../atoms/Tab/Tab.module.css';
import type { ITabGroup } from './ITabGroup';

export const tabGroupData = (): ITabGroup.ITabGroupData[] => {
  return [
    {
      id: '1',
      svg: <ImagePoll className={styles.icon} data-testid="image-poll-icon" />,
      value: 'Image Poll',
      postType: EPostType.ImagePoll,
    },
    {
      id: '2',
      svg: <TextPoll className={styles.icon} />,
      value: 'Text Poll',
      postType: EPostType.TextPoll,
    },
    {
      id: '3',
      svg: <MiniSurvey className={styles.icon} />,
      value: 'Mini Survey',
      postType: EPostType.MiniSurvey,
    },
  ];
};
