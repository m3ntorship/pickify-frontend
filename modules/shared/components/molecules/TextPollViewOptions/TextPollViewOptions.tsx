import React from 'react';
import type { FC, ReactElement } from 'react';
import TextOptionViewCovered from '../../atoms/TextOptionViewCovered/TextOptionViewCovered';
import TextOptionViewUncoverd from '../../atoms/TextOptionViewUncoverd/TextOptionViewUncoverd';
import type { ITextPollViewOptions } from './ITextPollViewOptions';
import { getVotesResults } from '../../../logic/votesLogic/votesLogic';

const OptionGroup: FC<ITextPollViewOptions.IProps> = ({
  optionsGroups,
  onOptionClick,
  optionCheckedId,
}): ReactElement => {
  const firstGroup = 0;
  const minusOne = -1;
  const { mostAndLeastVoted, optionsPercentage } = getVotesResults(
    optionsGroups.groups[firstGroup].options,
  );
  const alphabet: string[] = 'abcdefghijklmnopqrstuvwxyz'.split('');
  return (
    <div className="grid-rows-1 space-y-2">
      {optionsGroups.groups[firstGroup].options.map((option, index) => {
        const letter = alphabet[index];
        const { vote_count } = option;
        return (
          <div key={option.id} data-testid="option">
            {vote_count !== minusOne ? (
              <TextOptionViewUncoverd
                id={option.id}
                optionBody={option.body}
                isOptionChecked={optionCheckedId === option.id}
                letter={letter}
                mostVoted={option.vote_count === mostAndLeastVoted[firstGroup]}
                percentage={optionsPercentage[index]}
              />
            ) : (
              <TextOptionViewCovered
                id={option.id}
                onOptionClick={onOptionClick}
                optionBody={option.body}
                letter={letter}
              />
            )}
          </div>
        );
      })}
    </div>
  );
};
export default OptionGroup;
