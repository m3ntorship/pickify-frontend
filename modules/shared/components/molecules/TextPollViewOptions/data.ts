export const apiDummyData = {
  caption: 'nalyzing Delaware Frozen',
  options_groups: {
    groups: [
      {
        id: '03644270-7171-4147-b5a1-4233ff547f7a',
        name: 'Group Name',
        options: [
          {
            id: 'eeb0a32a-1a3c-4c7f-9e50-dd3d3f70b53d',
            body: 'ali tramsi',
          },
          {
            id: 'eeb0a32a-1a3c-4c7f-9e50-dd3d3f70b5aa',
            body: ' ahmed essam',
          },
          {
            id: 'eeb0a32a-1a3c-4c7f-9e50-dd3d3f70b53c',
            body: ' omar gamal',
          },
        ],
      },
    ],
  },
};
