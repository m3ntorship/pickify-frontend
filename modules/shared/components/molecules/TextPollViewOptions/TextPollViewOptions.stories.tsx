import type { ReactElement } from 'react';
import type { Story } from '@storybook/react';
import TextPollViewOptions from './TextPollViewOptions';
import type { ITextPollViewOptions } from './ITextPollViewOptions';
import { apiDummyData } from './data';

export default {
  title: 'Molecules/TextPollViewOptions',
  component: TextPollViewOptions,
  argTypes: {},
};

const Template: Story<ITextPollViewOptions.IProps> = (args): ReactElement => (
  <TextPollViewOptions {...args} />
);
export const Default = Template.bind({});
Default.args = {
  optionsGroups: apiDummyData.options_groups,
};
