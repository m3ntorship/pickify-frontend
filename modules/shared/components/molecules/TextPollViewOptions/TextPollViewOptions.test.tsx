import * as React from 'react';
import { render, screen } from '@testing-library/react';
import * as renderer from 'react-test-renderer';
import type { TargetElement } from '@testing-library/user-event';
import TextPollViewOptions from './TextPollViewOptions';
import { apiDummyData } from './data';

describe('TextPollViewOptions component', () => {
  it('should render () OptionGroup when we apply ()', () => {
    const mockedFn = jest.fn();
    const tree = renderer
      .create(
        <TextPollViewOptions
          optionCheckedId="1"
          isOptionChecked
          onOptionClick={mockedFn}
          optionsGroups={apiDummyData.options_groups}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render 3 Options when we apply options 3 length', () => {
    const mockedFn = jest.fn();
    const numberOfOptions = 3;

    render(
      <TextPollViewOptions
        optionCheckedId="1"
        isOptionChecked
        onOptionClick={mockedFn}
        optionsGroups={apiDummyData.options_groups}
      />,
    );

    const option: TargetElement[] = screen.getAllByTestId('option');

    expect(option).toHaveLength(numberOfOptions);
  });
});
