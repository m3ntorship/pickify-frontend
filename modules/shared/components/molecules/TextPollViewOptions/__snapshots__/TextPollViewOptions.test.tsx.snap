// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextPollViewOptions component should render () OptionGroup when we apply () 1`] = `
<div
  className="grid-rows-1 space-y-2"
>
  <div
    data-testid="option"
  >
    <button
      className="btnBody leastVoted"
      data-testid="eeb0a32a-1a3c-4c7f-9e50-dd3d3f70b53d"
      disabled={true}
      id="eeb0a32a-1a3c-4c7f-9e50-dd3d3f70b53d"
      type="button"
    >
      <span
        className="flex-container"
      >
        <span
          className="letter"
        >
          a
        </span>
        <p
          className="option"
        >
           
          ali tramsi
           
        </p>
        <span />
      </span>
      <span
        className="percentage"
      >
        
        22
        %
      </span>
    </button>
  </div>
  <div
    data-testid="option"
  >
    <button
      className="btnBody leastVoted"
      data-testid="eeb0a32a-1a3c-4c7f-9e50-dd3d3f70b5aa"
      disabled={true}
      id="eeb0a32a-1a3c-4c7f-9e50-dd3d3f70b5aa"
      type="button"
    >
      <span
        className="flex-container"
      >
        <span
          className="letter"
        >
          b
        </span>
        <p
          className="option"
        >
           
           ahmed essam
           
        </p>
        <span />
      </span>
      <span
        className="percentage"
      >
        
        33
        %
      </span>
    </button>
  </div>
  <div
    data-testid="option"
  >
    <button
      className="btnBody mostVoted"
      data-testid="eeb0a32a-1a3c-4c7f-9e50-dd3d3f70b53c"
      disabled={true}
      id="eeb0a32a-1a3c-4c7f-9e50-dd3d3f70b53c"
      type="button"
    >
      <span
        className="flex-container"
      >
        <span
          className="letter"
        >
          c
        </span>
        <p
          className="option"
        >
           
           omar gamal
           
        </p>
        <span />
      </span>
      <span
        className="percentage"
      >
        <svg
          className="goldenStarIcon"
          data-file-name="SvgGoldenStar"
        />
        44
        %
      </span>
    </button>
  </div>
</div>
`;
